papis/config.py

```diff
@@ -134,7 +134,7 @@ def initialize(self) -> None:
         configpy = get_configpy_file()
         if os.path.exists(configpy):
             with open(configpy) as fd:
-                exec(fd.read())
+                exec(fd.read(), globals())  # add user's defs to global name space
```

Comment:
I meant here to explain *why* this is needed, e.g. a usecase, not what adding `globals()` does. What did you need to add to the global name space?

Reply:
Hm, I should really read the description more carefully :cry: 

> This is necessary to allow any reasonably complicated custom function to be registered with the Jinja2Formatter environment (easily, at least). Executing config.py in the top level namespace ensures all functions defined within are discoverable by the formatter. This is very much simpler than any alternative.

What exactly do you mean here? This works for me as well
```python
from papis.format import Jinja2Formatter

def and_others(s: str) -> str:
    return f"{s} and others"


env = Jinja2Formatter.get_environment()
env.filters["and_others"] = and_others
```

Reply:
Haha, you've made your point, both my long and short-form explanations are inadequate. Let's think how best to make this clear.

Try the following config without globals(), it should fail with a name not found error.

```
from papis.format import Jinja2Formatter

def _my_own_fstring(s: str) -> str:
    return f"{s} and others"

def and_others(s: str) -> str:
    return _my_own_fstring(s)

env = Jinja2Formatter.get_environment()
env.filters["and_others"] = and_others
```

Reply:
Ah, I see! :facepalm: